autoload/textobj/variable_segment.vim

```diff
@@ -1,8 +1,11 @@
 function! s:select(object_type, right_boundary)
-    let left_boundaries = ['_\+\i', '\<', '\l\u', '\u\u\ze\l', '\a\d', '\d\a']
+    let left_boundaries = ['_\+\i', '-\+\i', '\<', '\l\u', '\u\u\ze\l', '\a\d', '\d\a']
     call search(join(left_boundaries, '\|'), 'bce')
     let start_position = getpos('.')
 
+    let orig_iskeyword=&l:iskeyword
```

Comment:
This looks suspicious to me.

I don't know that I'm thinking clearly enough, but I'd expect this to be done outside of this plugin entirely -- i.e., I'd expect this plugin to rely on a properly set `iskeyword` for the filetype that it's operating on, so if that ftplugin uses dashes, then we rely on that.

Perhaps you're doing this just to satisfy the test case you wrote (which by the way thanks!, well done on those)